src/Events/ApplicationEvent.php

```diff
@@ -26,21 +27,35 @@ public function handle($event): void
             }
 
             if ($breadcrumbEnabled) {
-                Honeybadger::addBreadcrumb($payload->message, $payload->metadata, $payload->type);
+                Honeybadger::addBreadcrumb($payload->message, $payload->metadata, $payload->category);
             }
 
             if ($eventEnabled) {
-                $merged = array_merge($payload->metadata, [
-                    'event_type' => $payload->type,
-                    'message' => $payload->message
-                ]);
-                Honeybadger::event($merged);
+                // set requestId, if available
+                $logContext = Log::sharedContext();
+                if (isset($logContext['requestId']) && !isset($payload->metadata['requestId'])) {
+                    $payload->metadata['requestId'] = $logContext['requestId'];
+                }
+                Honeybadger::event($payload->type, $payload->metadata);
             }
         } catch (\Throwable $e) {
             // Do nothing; we shouldn't crash the user's app for this.
         }
     }
 
+    protected function getDurationInMs(float $startTime): ?string
+    {
+        if (!isset($startTime)) {
+            return null;
+        }
+
+        $duration = microtime(true) - $startTime;
+
+        // Insights adds three zeros (000) to the number if I add the 'ms' suffix ¯\_(ツ)_/¯
```

Comment:
@rabidpraxis Can you help me out with this one?
I saw you were adding 'ms' to this variable, but when I'm testing it's adding 3 zeroes on Insights. For example, the duration in the screenshot attached should be 1465 (milliseconds).
![CleanShot 2024-09-07 at 18 09 49@2x](https://github.com/user-attachments/assets/057c76ea-7602-4842-b3d4-a601396bf7cc)

Reply:
That one also threw me for a loop as I was building out dashboard test data.

Our formatter that converts string fields to durations (e.g. "5ms") normalizes the integer value into microseconds :)

Reply:
Oh I see. How do you think we should go about this then?
Should we use ms or microseconds for duration? Personally, I'm leaning towards ms simply because this is what I usually see in similar services.
Additionally, do you think it would be a good idea to be explicit about the metric unit, like "durationMs"?

Reply:
Let's stick with sending duration as a millisecond string with "ms" suffix. 

When the formatter extracts the time value, we store the unit in the event metadata. With this, we have the potential to automatically display the duration in a more friendly format on the front end (`3.2s` vs `3243200`).

No, I don't think we should include the unit in the field name. I would love to get to a state where a user can construct a query without being concerned about the recorded unit, allowing for a query like `filter duration > 1s`.

Reply:
> Let's stick with sending duration as a millisecond string with "ms" suffix.
👍 

> No, I don't think we should include the unit in the field name. I would love to get to a state where a user can construct a query without being concerned about the recorded unit, allowing for a query like filter duration > 1s.
👍 This would be a neat feature 👌

Reply:
Just to confirm, the following screenshot should read duration as 283000 microseconds right?
![CleanShot 2024-09-14 at 16 39 23@2x](https://github.com/user-attachments/assets/b37b4242-4f77-49a4-ad7b-54256072017d)

Reply:
Correct.